### PR TITLE
designed api database

### LINKED ISSUE #1: Design the Database
Design the ecommerce database for use in the API

### PR DESCRIPTION
**Pull request by robterrins**
resolves #1  - design database

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# SERVER CONFIGURATION
+PORT=4000
+SESSION_SECRET='session_secret'
```
Comment (erikgalajda) [inline .env:3]: sensitive information to be stored in git, .env should not be included

File: database.psql [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+CREATE TABLE IF NOT EXISTS carts (
+  id SERIAL PRIMARY KEY,
+  product_id INTEGER,
+  created TIMESTAMP,
+  modified TIMESTAMP
+);
+
+CREATE TABLE IF NOT EXISTS orders_products (
+  order_id INTEGER,
+  product_id INTEGER,
```
Comment (erikgalajda) [inline database.psql:57]: should be composite primary key

File: database.psql [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  quantity INTEGER
+);
+
+CREATE TABLE IF NOT EXISTS carts_products (
+  cart_id INTEGER,
+  product_id INTEGER,
```
Comment (erikgalajda) [inline database.psql:63]: should be composite primary key

**Review by erikgalajda — COMMENTED** [review]
table users and customer_details have both first_name, last_name columns, i think only one table should have these columns

**Review by erikgalajda — COMMENTED** [review]
what was your intention when connecting customer_details with users table? According the schema it looks like one customer_details entry can belong to many users.

**Review by erikgalajda — COMMENTED** [review]
in my design i avoided the connection between users and orders and instead i linked orders to carts, what do you think?

**Comment by robterrins** [comment]
> what was your intention when connecting customer_details with users table? According the schema it looks like one customer_details entry can belong to many users.

That was not my intention, the single arrow line was supposed to mean 1-to-1. Although now I realise that it doesnt work for the products-orders-carts relationships

**Comment by robterrins** [comment]
> in my design i avoided the connection between users and orders and instead i linked orders to carts, what do you think?

I think the link to orders and carts is fine but I would include a link back to the users aswell via foreign key

**Pull request merged**